nolabs/modules/biobuddy/send_message_feature.py

```diff
@@ -37,40 +39,54 @@ def handle(self, request: SendMessageRequest) -> SendMessageResponse:
-
-            print(assistant_message)
-            if assistant_message.tool_calls and assistant_message.tool_calls.actual_instance:
-                function_name = assistant_message.tool_calls.actual_instance[0]['function']['name']
-                arguments = json.loads(assistant_message.tool_calls.actual_instance[0]['function']['arguments'])
-                if function_name in self._functions:
-                    function_call = self._functions[function_name].execute(experiment_id=experiment_id,
-                                                                           arguments=arguments)
-                    self._file_management.update_conversation(experiment_id,
-                                                              Message(role='user',
-                                                                      message=RegularMessage(
-                                                                          content=request.message_content
-                                                                      ),
-                                                                      type='text')
-                                                              )
-                    self._file_management.update_conversation(experiment_id,
-                                                              Message(role='assistant',
-                                                                      message=FunctionCall(
-                                                                          function_name=function_call.function_name,
-                                                                          parameters=[FunctionParam(name=param.name,
-                                                                                                    value=param.value) for param in
-                                                                                      function_call.parameters]
-                                                                                           ),
-                                                                      type="function")
-                                                              )
-                    return SendMessageResponse(biobuddy_response=ApiMessage(role='assistant',
-                                                                            message=ApiFunctionCall(**dataclasses.asdict(function_call)),
-                                                                            type="function")
-                                               )
+                send_message_to_bio_buddy_request=biobuddy_request)
+
+            if assistant_message.reply_type == "function_calls":
+                print(assistant_message.content)
+                function_calls = ast.literal_eval(assistant_message.content)
+                print(function_calls)
+                self._file_management.update_conversation(experiment_id,
```

Comment:
Queries that modify the data are not the good decision. But we'll skip it to later once we go to DDD there would be a domain event like BioBuddyMessageSent